os/arch/arm/src/armv7-r/arm_assert.c

```diff
@@ -918,6 +931,67 @@ static void up_dumpstate(void)
 static void _up_assert(int errorcode) noreturn_function;
 static void _up_assert(int errorcode)
 {
+#ifdef CONFIG_FAULT_MGR
+	int ret;
+	mqd_t fault_queue;
+	pid_t pid;
+	uint32_t assert_pc;
+	uint32_t ksram_segment_end = (uint32_t)__ksram_segment_start__ + (uint32_t)__ksram_segment_size__;
+	uint32_t kflash_segment_end = (uint32_t)__kflash_segment_start__ + (uint32_t)__kflash_segment_size__;
+
+	/* Extract the PC value of instruction which caused the abort/assert */
+
+	if (current_regs) {
+		assert_pc = current_regs[REG_LR];
+	} else {
+		assert_pc = (uint32_t)g_assertpc[0];
+	}
+
+	/* Check if the PC lies within kernel context */
+
+	if ((assert_pc >= (uint32_t)__ksram_segment_start__ && assert_pc <= ksram_segment_end) || (assert_pc > (uint32_t)__kflash_segment_start__ && assert_pc < kflash_segment_end)) {
+
+		/* Reboot the system if fault is in kernel context */
```

Comment:
This case is same as original. 
```
#ifdef config_falut
if kernel?
   reset
else
   send message to fault manager
#else 
  rest
#endif 
```
How about this?
```
#ifdef config_fault
  if !kernel
     send message to fault manager
  else
#endif 
  reset (original)
```

Reply:
There is some difference in else case and original logic.
In fault manager else case, we are rebooting the system, and without fault manager original case used to loop infinitely.
Should we change the original logic from loop infinitely to reboot?

Reply:
There is little difference between this case and original one.
In fault manager else part, we are rebooting the board, and in original case, we are looping infinitely. 
Original case checks 3 conditions which is also little different then the condition check of fault manager.
What is your opinion in this?

Reply:
> In fault manager else part, we are rebooting the board, and in original case, we are looping infinitely.

`CONFIG_BOARDCTL_RESET` means this board supports reboot by sw. It does not mean auto reset on assert.
Let's make a new config to make user decide auto reset on assert.
This is a request of assert module. But fault manager also uses that code. That's why I request it now.
Or, you can push it as of now. Later we can modify that with another commit.

Reply:
In this commit, you give an option to select infinite loop or auto reset in kernel assert.
But there is no way to select inifinite loop for user assert.

Reply:
In case of kernel assert, in this commit, it will go into infinite loop only if  CONFIG_BOARDCTL_RESET is not set.

As per the design we wanted to restart the user application in case of fault, so for the registered apps we have not given infinite loop option, but if the application is not registered then it will be handle the same way as kernel assert.

So if it is fine, we can push it this way in this commit, later we can modify if another scenario is needed with other commit.

Reply:
I have added the feature to select infinite loop in fault manager
please check